onnx/defs/tensor/defs.cc

```diff
@@ -1595,6 +1664,28 @@ ONNX_OPERATOR_SET_SCHEMA(
                                             : // i - axis < q
                     data_shape.dim(i - q + 1); // i < out_rank < q + r - 1
           }
+        })
+        .PartialDataPropagationFunction([](DataPropagationContext& ctx) {
+          if (!axisIsZero(ctx, true)) {
+            return;
+          }
+          const auto input_data = ctx.getInputData(0);
+          const auto input_indices = ctx.getInputData(1);
+          if (input_data == nullptr || input_indices == nullptr) {
+            return;
+          }
+          TensorShapeProto tsp;
+          for (int i = 0; i < input_indices->dim_size(); ++i) {
+            if (input_indices->dim(i).has_dim_value()) {
+              int index = input_indices->dim(i).dim_value();
+              if (index < input_data->dim_size()) {
```

Comment:
I think that in the else case we should return (instead of creating an incorrect OutputData). (While we could produce an error/exception, it may be simpler to not bother about that for now.

Reply:
The index value is allowed to be negative to count backwards.

Reply:
Thanks for catching these 2 issues. Both updated